src/tools/tidy/src/deps.rs

```diff
@@ -24,52 +28,201 @@ static LICENSES: &'static [&'static str] = &[
+    "cssparser-macros",   // MPL-2.0, rustdoc
+    "selectors",          // MPL-2.0, rustdoc
+    "clippy_lints",       // MPL-2.0 rls
+];
+
+/// Which crates to check against the whitelist?
+static WHITELIST_CRATES: &'static [Crate] =
+    &[Crate("rustc", "0.0.0"), Crate("rustc_trans", "0.0.0")];
+
+/// Whitelist of crates rustc is allowed to depend on. Avoid adding to the list if possible.
+static WHITELIST: &'static [Crate] = &[
+//    Crate("ar", "0.3.1"),
+//    Crate("arena", "0.0.0"),
+//    Crate("backtrace", "0.3.5"),
+//    Crate("backtrace-sys", "0.1.16"),
+//    Crate("bitflags", "1.0.1"),
+//    Crate("build_helper", "0.1.0"),
+//    Crate("byteorder", "1.2.1"),
+//    Crate("cc", "1.0.4"),
+//    Crate("cfg-if", "0.1.2"),
+//    Crate("cmake", "0.1.29"),
+//    Crate("filetime", "0.1.15"),
+//    Crate("flate2", "1.0.1"),
+//    Crate("fmt_macros", "0.0.0"),
+//    Crate("fuchsia-zircon", "0.3.3"),
+//    Crate("fuchsia-zircon-sys", "0.3.3"),
+//    Crate("graphviz", "0.0.0"),
+//    Crate("jobserver", "0.1.9"),
+//    Crate("kernel32-sys", "0.2.2"),
+//    Crate("lazy_static", "0.2.11"),
+//    Crate("libc", "0.2.36"),
+//    Crate("log", "0.4.1"),
+//    Crate("log_settings", "0.1.1"),
+//    Crate("miniz-sys", "0.1.10"),
+//    Crate("num_cpus", "1.8.0"),
+//    Crate("owning_ref", "0.3.3"),
+//    Crate("parking_lot", "0.5.3"),
+//    Crate("parking_lot_core", "0.2.9"),
+//    Crate("rand", "0.3.20"),
+//    Crate("redox_syscall", "0.1.37"),
+//    Crate("rustc", "0.0.0"),
+//    Crate("rustc-demangle", "0.1.5"),
+//    Crate("rustc_allocator", "0.0.0"),
+//    Crate("rustc_apfloat", "0.0.0"),
+//    Crate("rustc_back", "0.0.0"),
+//    Crate("rustc_binaryen", "0.0.0"),
+//    Crate("rustc_const_eval", "0.0.0"),
+//    Crate("rustc_const_math", "0.0.0"),
+//    Crate("rustc_cratesio_shim", "0.0.0"),
+//    Crate("rustc_data_structures", "0.0.0"),
+//    Crate("rustc_errors", "0.0.0"),
+//    Crate("rustc_incremental", "0.0.0"),
+//    Crate("rustc_llvm", "0.0.0"),
+//    Crate("rustc_mir", "0.0.0"),
+//    Crate("rustc_platform_intrinsics", "0.0.0"),
+//    Crate("rustc_trans", "0.0.0"),
+//    Crate("rustc_trans_utils", "0.0.0"),
+//    Crate("serialize", "0.0.0"),
+//    Crate("smallvec", "0.6.0"),
+//    Crate("stable_deref_trait", "1.0.0"),
+//    Crate("syntax", "0.0.0"),
+//    Crate("syntax_pos", "0.0.0"),
+//    Crate("tempdir", "0.3.5"),
+//    Crate("unicode-width", "0.1.4"),
+//    Crate("winapi", "0.2.8"),
+//    Crate("winapi", "0.3.4"),
+//    Crate("winapi-build", "0.1.1"),
+//    Crate("winapi-i686-pc-windows-gnu", "0.4.0"),
+//    Crate("winapi-x86_64-pc-windows-gnu", "0.4.0"),
 ];
 
+// Some types for Serde to deserialize the output of `cargo metadata` to...
+
+#[derive(Deserialize)]
+struct Output {
+    resolve: Resolve,
+
+    // Not used, but needed to not confuse serde :P
+    #[allow(dead_code)] packages: Vec<Package>,
```

Comment:
Could this get removed? If not, how come?

Reply:
done